CMakeLists.txt

```
@@ -30,7 +30,7 @@ set(CMAKE_CONFIGURATION_TYPES
 
 # CMake looks for these variables to be defined for the custom configuration.
 set(CMAKE_CXX_FLAGS_RELEASEWITHDOCS
-    ${CMAKE_CXX_FLAGS_RELEASE}
+    "${CMAKE_CXX_FLAGS_RELEASE} -O3"
```

```suggestion
    "-O3 -DNDEBUG"
```
`-DNDEBUG` makes `assert` noops and may deactivate other debug code paths depending on the library.

I recommend making these explicit here. The reason using `${CMAKE_CXX_FLAGS_RELEASE}` didn't work to set this is likely because that variable is not populated with its defaults until later during the configure step.

`-O3 -DNDEBUG` is the default for gcc/clang type compilers. This may not be valid for MSVC. You may want to add a conditional based on the compiler to set optimizations in the windows builds as well.

---

Good idea. I observed DNDEBUG in my build but making it explicit here is probably safer.

---

Made the changes locally just to be sure I wouldn't observe any changes. It looks fine to me.